packages/scandipwa/src/route/ProductPage/ProductPage.component.js

```diff
@@ -132,31 +164,101 @@ export class ProductPage extends PureComponent {
         );
     }
 
-    renderAdditionalSections() {
+    renderProductInformationTab() {
         const {
             dataSource,
             parameters,
+            areDetailsLoaded,
+            isInformationTabEmpty
+        } = this.props;
+
+        if (isInformationTabEmpty) {
+            return null;
+        }
+
+        return (
+            <ProductInformation
+              product={ { ...dataSource, parameters } }
+              areDetailsLoaded={ areDetailsLoaded }
+            />
+        );
+    }
+
+    renderProductAttributesTab() {
+        const {
+            dataSource,
+            parameters,
+            areDetailsLoaded,
+            isAttributesTabEmpty
+        } = this.props;
+
+        if (isAttributesTabEmpty) {
+            return null;
+        }
+
+        return (
+            <ProductAttributes
+              product={ { ...dataSource, parameters } }
+              areDetailsLoaded={ areDetailsLoaded }
+            />
+        );
+    }
+
+    renderProductReviewsTab() {
+        const {
+            dataSource,
+            areDetailsLoaded
+        } = this.props;
+
+        return (
+            <ProductReviews
+              product={ dataSource }
+              areDetailsLoaded={ areDetailsLoaded }
+            />
+        );
+    }
+
+    renderProductTabItems() {
+        const productTabItems = [];
+        Object.values(this.tabMap).map((item) => {
+            productTabItems.push(item.render());
+            return true;
+        });
+
+        return productTabItems.filter(Boolean).map((item) => <div key={ item.toString() }>{ item }</div>);
+    }
+
+    getTabNames() {
+        const renderTabs = [];
+
+        Object.values(this.tabMap).map((item) => {
+            if (!item.shouldTabRender()) {
+                renderTabs.push(item.name);
+            }
+
+            return true;
+        });
```

Comment:
Please learn what each of this functions do and how to properly use it. https://docs.scandipwa.com/building-your-app/javascript-code-style#working-with-arrays

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array <- this one official